nixos/modules/services/development/livebook.nix

```diff
@@ -0,0 +1,101 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.livebook;
+in
+{
+  options.services.livebook = {
+    enableUserService = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Whether to enable a user service for Livebook.  This implies
+        `installUserService`.
+      '';
+    };
+
+    installUserService = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Whether to install a user service for Livebook.  Unless
+        `enableUserService` is also set, users will have to manually activate
+        the service with `systemctl --user enable livebook`.
+      '';
+    };
+
+    environment = mkOption {
+      type = with types; attrsOf str;
+      description = lib.mdDoc ''
+        The environment to pass to the Livebook service.  You need to set at least
+        `LIVEBOOK_PASSWORD`.  See `livebook server --help` for other options.
+      '';
+      example = {
+        LIVEBOOK_PASSWORD = "a password at least 30 characters long";
+        LIVEBOOK_PORT = 30123;
+      };
+    };
+
+    erlang_node_short_name = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      example = "livebook";
+      description = "A short name for the distributed node.";
+    };
+
+    erlang_node_name = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      example = "livebook@127.0.0.1";
+      description = "The name for the app distributed node.";
+    };
+
+    port = mkOption {
+      type = with types; nullOr port;
+      default = null;
+      example = 30123;
+      description = "The port to start the web application on.";
+    };
+
+    cookie = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      description = "The cookie for the app distributed node.";
+    };
+
+    address = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      description = lib.mdDoc ''
+        The address to start the web application on.  Must be a valid IPv4 or
+        IPv6 address.
+      '';
+    };
```

Comment:
What happens if these are `null`? Do they have some kind of internal default? Maybe we should mirror the defaults externally and make them non nullable, so the user can discover the defaults in the manual more easily?

Reply:
I've added the upstream defaults to `port` and `address`.

The other options don't have defaults.  If you don't set them, then the Erlang node doesn't have distribution enabled.  I don't think we need to document this here because anyone who wants a distributed node will recognize what these options are.